### ISSUE TITLE
Installation Failure: This script contains malicious content and has been blocked by your antivirus software.

### ISSUE THREAD

**Issue by DeeBee20** [fireeye/commando-vm#166]
**Description**
Spun up a fresh virtual machine. carried out all updates, disabled all firewalls, windows defender, anti-tamper etc, took snapshot of virtual machine. 

**Operating System**
 - OS: [Win10x64 Pro 1909]
 - HDD: [100GB]
 - RAM: [12GB]

**Followed the standard install process**
```
Standard install

    Create and configure a new Windows Virtual Machine

    Ensure VM is updated completely. You may have to check for updates, reboot, and check again until no more remain

    Take a snapshot of your machine!
    Download and copy install.ps1 on your newly configured machine.
    Open PowerShell as an Administrator
    Unblock the install file by running Unblock-File .\install.ps1
    Enable script execution by running Set-ExecutionPolicy Unrestricted -f
    Finally, execute the installer script as follows:

    .\install.ps1
    You can also pass your password as an argument: .\install.ps1 -password <password>

The script will set up the Boxstarter environment and proceed to download and install the Commando VM environment. You will be prompted for the administrator password in order to automate host restarts during installation. If you do not have a password set, hitting enter when prompted will also work.
```
**Error Output**
Below is the error output
```
Chocolatey upgraded 1/1 packages. 0 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
+ Boxstarter finished Calling Chocolatey to install Invoke-Obfuscation.fireeye. This may take several minutes to complete... 00:00:07.8180351
+ Boxstarter finished Calling Chocolatey to install Invoke-Obfuscation.fireeye. This may take several minutes to complete... 00:00:07.8180351

2020/09/10 23:01:08 [Invoke-Obfuscation.fireeye] chocolateyinstall.ps1 [+] INFO : Install of Invoke-Obfuscation.fireeye finished successfully
2020/09/10 23:01:08 [Invoke-Obfuscation.fireeye] chocolateyinstall.ps1 [+] INFO : Install of Invoke-Obfuscation.fireeye finished successfully
ERROR: At line:1 char:1
+ Write-Host 2020/09/10 23:01:08 [Invoke-Obfuscation.fireeye] chocolate ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
The install of commandovm.win10.installer.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\commandovm.win10.installer.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - commandovm.win10.installer.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\commandovm.win10.installer.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.
++ Boxstarter finished Calling Chocolatey to install commandovm.win10.installer.fireeye. This may take several minutes to complete... 00:11:52.0401089
True
Boxstarter: Restore Automatic Updates from Windows Update
Boxstarter: Enabling UAC
Type ENTER to exit:
```
**Log Output**
Below is the ending of the log output
```
2020-09-10 22:10:28,870 8820 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2".
2020-09-10 22:10:28,884 8820 [DEBUG] - There was no original file at 'C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2\.files'
2020-09-10 22:10:28,884 8820 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2\.extra".
2020-09-10 22:10:28,884 8820 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2\.version".
2020-09-10 22:10:28,884 8820 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2\.sxs".
2020-09-10 22:10:28,884 8820 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\commandovm.win10.preconfig.fireeye.2.0.0.2\.pin".
2020-09-10 22:10:28,884 8820 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2020-09-10 22:10:28,884 8820 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\commandovm.win10.preconfig.fireeye\.chocolateyPending".
2020-09-10 22:10:28,884 8820 [INFO ] -  The upgrade of commandovm.win10.preconfig.fireeye was successful.
2020-09-10 22:10:28,901 8820 [INFO ] -   Software install location not explicitly set, could be in package or
  default install location if installer.
2020-09-10 22:10:28,901 8820 [DEBUG] - Attempting to delete file "C:\Users\Commando\AppData\Local\NuGet\Cache\commandovm.win10.preconfig.fireeye.2.0.0.2.nupkg".
2020-09-10 22:10:28,901 8820 [WARN ] - 
Chocolatey upgraded 1/1 packages. 
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2020-09-10 22:10:28,932 8820 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2020-09-10 22:10:28,963 8820 [DEBUG] - Exiting with 0
```
**Next Steps**
Any information on what has gone wrong or how i can fix this please.

**Comment by day1player** [comment]
@DeeBee20 it looks like Tamper Protection may not have been disabled completely and Defender was turned back on mid-install based on this line in your output:
```
This script contains malicious content and has been blocked by your antivirus software.
```

I recommend going in and manually disabling both Tamper Protection _and_ Defender and then reattempting install

**Comment by DeeBee20** [comment]
I have gone trhough disabling all Windows 10 security features, an still had no luck with this. How do i get added to the bloodhound slack channel so i can work through these issues please.

**Comment by day1player** [comment]
@DeeBee20 can you send the error output now? If you are no longer receiving the error for malicious content, are you receiving any errors?

**Comment by DeeBee20** [comment]
I am just running through it again now to get the output, I got the ending product with desktop changed and files and background as should be, but then docker would not work, covenant, amass and spiderfoot had all not installed.

Possibly maybe something i am doing but then i feel it could be the scripts are having faults, i'm a little losted at the moment with it.

**Comment by DeeBee20** [comment]
```
System.InvalidOperationException:
Failed to set version to docker-desktop: exit code: -1
 stdout: The attempted operation is not supported for the type of object referenced.

 stderr: 
   at Docker.Core.WslShortLivedCommandResult.LogAndThrowIfUnexpectedExitCode(String prefix, ILogger log, Int32 expectedExitCode) in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Core\WslCommand.cs:line 119
   at Docker.Engines.WSL2.WSL2Provisioning.<ProvisionAsync>d__8.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\WSL2Provisioning.cs:line 82
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__24.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\LinuxWSL2Engine.cs:line 94
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92
```

**Comment by DeeBee20** [comment]
```
covenant.fireeye package files install completed. Performing other installation steps.
Refreshing environment variables from the registry for powershell.exe. Please wait...
Finished
Downloading Covenant.fireeye
  from 'https://github.com/cobbr/Covenant/archive/02e8499a4c94fe7a74250bd20049420521621b32.zip'

Download of Covenant-02e8499a4c94fe7a74250bd20049420521621b32.zip (-1 B) completed.
Hashes match.
Extracting C:\Users\Commando\AppData\Local\Temp\Covenant.fireeye\1.0.0.6\Covenant-02e8499a4c94fe7a74250bd20049420521621b32.zip to C:\ProgramData\chocolatey\lib\Covenant.fireeye\tools\temp_563bab6d-aea2-4fcb-a830-0d496e69b4e8...
C:\ProgramData\chocolatey\lib\Covenant.fireeye\tools\temp_563bab6d-aea2-4fcb-a830-0d496e69b4e8
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Operation did not complete successfully because the file contains a virus or potentially unwanted software.

Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
The install of covenant.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\Covenant.fireeye\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - covenant.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\Covenant.fireeye\tools\chocolateyInstall.ps1'.
 See log for details.
```

**Comment by DeeBee20** [comment]
```
amass.docker.fireeye package files install completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: time="2020-09-14T20:02:54+01:00" level=error msg="Can't add file \\\\?\\C:\\Users\\Commando\\AppData\\Local\\Temp\\docker-build-git689909683\\.git\\hooks\\pre-receive.sample to tar: io: read/write on closed pipe"
The install of amass.docker.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\amass.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - amass.docker.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\amass.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.
COMMANDO 9/14/2020 8:02:57 PM
```

**Comment by DeeBee20** [comment]
```
lonelypotato.fireeye package files install completed. Performing other installation steps.
Attempt to get headers for https://github.com/decoder-it/lonelypotato/archive/1845a19662856abd1e3f84f3ebdb9c029e93ab8d.zip failed.
  The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://github.com/decoder-it/lonelypotato/archive/1845a19662856abd1e3f84f3ebdb9c029e93ab8d.zip'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
Downloading lonelypotato.fireeye
  from 'https://github.com/decoder-it/lonelypotato/archive/1845a19662856abd1e3f84f3ebdb9c029e93ab8d.zip'
ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://github.com/decoder-it/lonelypotato/archive/1845a19662856abd1e3f84f3ebdb9c029e93ab8d.zip'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
The install of lonelypotato.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\lonelypotato.fireeye\Tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - lonelypotato.fireeye (exited 404) - Error while running 'C:\ProgramData\chocolatey\lib\lonelypotato.fireeye\Tools\chocolateyInstall.ps1'.
 See log for details.
COMMANDO 9/14/2020 8:05:33 PM
```

**Comment by DeeBee20** [comment]
```
spiderfoot.docker.fireeye package files install completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: time="2020-09-14T20:07:05+01:00" level=error msg="Can't add file \\\\?\\C:\\Users\\Commando\\AppData\\Local\\Temp\\docker-build-git604228391\\.git\\hooks\\update.sample to tar: io: read/write on closed pipe"
The install of spiderfoot.docker.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\spiderfoot.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - spiderfoot.docker.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\spiderfoot.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.
COMMANDO 9/14/2020 8:07:08 PM
```

**Comment by DeeBee20** [comment]
```
wfuzz.fireeye package files install completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop:     ERROR: Command errored out with exit status 1:
The install of wfuzz.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\wfuzz.fireeye\Tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - wfuzz.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\wfuzz.fireeye\Tools\chocolateyInstall.ps1'.
 See log for details.
COMMANDO 9/14/2020 8:28:25 PM
```

**Comment by day1player** [comment]
@DeeBee20 could you tell me what command you are running to install?

**Comment by DeeBee20** [comment]
well the default commands that are on the standard install, didn't work keeps erroring out, i had to use the following.
```
cd C:\Users\Commando\Downloads\commando-vm-master\commando-vm-master
Unblock-File .\install.ps1
Set-ExecutionPolicy Unrestricted -f
.\install.ps1 -nochecks 1
```

**Comment by DeeBee20** [comment]
which then had issues installing the commandovm.win10.installer.fireeye, so i had to use the command below to install that.
```
cinst -y commandovm.win10.installer.fireeye
```

**Comment by DeeBee20** [comment]
another issue during installation was Invoke-Obfuscation.fireeye.
```
cinst -y Invoke-Obfuscation.fireeye
```

**Comment by day1player** [comment]
@DeeBee20 It really looks like you are using an outdated installer script. Most of those packages were removed from the default install (Covenant, Docker, wfuzz, all docker packages).. 

If you would like to install docker and the associated packages I can definitely help as long as you are not using a Windows _host_ machine. Nested virtualization (docker inside of a VM) does not work on Windows (#94 ) host machines (it's possible, but I cant help you there, you can check the previous issue). As far as Covenant is concerned, it was removed some time ago, and wfuzz was removed in the last update. If you already finished the install you should be fine, there is no need to wipe and restart as far as I can remember. You should have most all of the tools and config changes set and working.

for Invoke-Obfuscation, please try again with the `-f` flag and if you get an error please let me know what the output is

**Comment by DeeBee20** [comment]
i've just downloaded the github zip, so the one on github must be outdated then, i was wanting to use this vm for a red teaming exam i have coming up as the main tool it had was covenant which is part of the red teaming course tools, so if your saying this is no longer included within commando vm i will just get rid of the vm and build my own, as this was the main reason i was wanting to use this vm.

**Comment by day1player** [comment]
@DeeBee20 I am not sure what happened, perhaps it was cached somewhere or somehow a different commit hash was downloaded for you but this is where all of the packages are set: https://github.com/fireeye/commando-vm/blob/master/commandovm.win10.installer.fireeye/tools/packages.json and there is no covenant, docker, wfuzz, etc. in there.. 

Try this link here to ensure - https://github.com/fireeye/commando-vm/archive/a2b96c9a60ffce41fb28fbc7f88ea354e754127c.zip

**Comment by day1player** [comment]
Also @DeeBee20 we do still offer automated docker install (its just not default anymore), you just have to specify a profile in the install arguments.. I have a few managed profiles you can check out here: https://github.com/fireeye/commando-vm/tree/master/Profiles

I would recommend installing with the nested v profile:
```
.\install.ps1 -profile_file .\Profiles\nestedv.json
```

**Comment by day1player** [comment]
That said if you want Covenant installed, you can do so by following the Covenant install instructions (do not use the covenant.fireeye package as it is no longer updated/supported). I am working on a Covenant docker package but have not finished it yet. 

I have used Covenant with Docker on Commando with great success

**Comment by DeeBee20** [comment]
@day1player ahhh ok, this sounds great, using the full.json profile does this include all the tools from the github main page less docker, covenant, wfuzz, amass, spiderfoot.

I can then use your docker nestedv.json to install docker.

An then i can just implement a covenant docker pull i presume or find covenant install for docker

**Comment by day1player** [comment]
@DeeBee20 Yeah it should be no problem, I just updated my above comment with a link to the docker install instructions for Covenant. 

if you use the full.json profile it will include Docker, amass and spiderfoot but it will not include covenant or wfuzz.. we used to support Covenant through dotnet (convenant.fireeye) but we had several issues with automating that install and I wanted to move to Docker (I just havent come up with a script for that yet) and then for wfuzz it actually doesnt appear it is possible to get working properly on [windows](https://github.com/fireeye/commando-vm/issues/86#issuecomment-536638912) so I removed it

**Comment by DeeBee20** [comment]
@day1player great news, massive help thank you very much. So i am correct in thinking for installing the full.json profile i just have to run.
```
Unblock-File .\install.ps1
Set-ExecutionPolicy Unrestricted -f
.\install.ps1 -profile_file .\Profiles\full.json
```

**Comment by day1player** [comment]
@DeeBee20 Yep you are correct in that assumption. Any issues just let me know. If you want to get into the BH slack just send me your email to commandovm@fireeye.com and I will request to have you added.

**Comment by DeeBee20** [comment]
@day1player Just one last thing how are you disabling the tamper protection, as i have it turned off in the security centre along with all other security features but for some reason i am still getting the warning as below.
```
127c> .\install.ps1 -profile_file .\Profiles\full.json


 ____________________________________________________________________________
|                                                                            |
|                                                                            |
|        _________                                           .___            |
|        \_   ___ \  ____   _____   _____ _____    ____    __| _/____        |
|        /    \  \/ /  _ \ /     \ /     \\__  \  /    \  / __ |/  _ \       |
|        \     \___(  <_> )  Y Y  \  Y Y  \/ __ \|   |  \/ /_/ (  <_> )      |
|         \______  /\____/|__|_|  /__|_|  (____  /___|  /\____ |\____/       |
|                \/             \/      \/     \/     \/      \/             |
|                       C O M P L E T E  M A N D I A N T                     |
|                            O F F E N S I V E   V M                         |
|                                                                            |
|                                 Version 2020.2                             |
|                             commandovm@fireeye.com                         |
|____________________________________________________________________________|
|                                                                            |
|                                   Created by                               |
|                            Jake Barteaux @day1player                       |
|                               Proactive Services                           |
|                         Blaine Stancill @MalwareMechanic                   |
|                                   Nhan Huynh                               |
|                    FireEye Labs Advanced Reverse Engineering               |
|____________________________________________________________________________|

[+] Using the following profile .\Profiles\full.json
WARNING: [+] You are using a custom profile and list of packages,
WARNING: [+] You will NOT receive new tools automatically when running choco update.
[-] Do you want to continue? Y/N Y
        Continuing...
[+] Checking if script is running as administrator..
        this is the way
[+] Checking to make sure Windows Defender Tamper Protection is disabled
[!] Please disable Windows Defender Tamper Protection and retry install.
        [+] Hint: https://www.tenforums.com/tutorials/123792-turn-off-tamper-protection-windows-defender-antivirus.html
[-] Do you need to change this setting? Y/N
```

**Comment by DeeBee20** [comment]
@day1player i have done it as shown at link.

 https://www.tenforums.com/tutorials/123792-turn-off-tamper-protection-microsoft-defender-antivirus.html

**Comment by day1player** [comment]
@DeeBee20 as long as you are using the GUI to disable tamper protection you should be fine. I could not reliably ensure tamper protection was disabled so thats why I made it so people could bypass that warning. 

If you're showing in the GUI that it is disabled, you are safe.

**Comment by DeeBee20** [comment]
@day1player Thank you, i am running through the new installation process now, i will post any issues i have. If i have any.

**Comment by DeeBee20** [comment]
@day1player  The following issues with the profile full.json install.
```
openjdk13 v13.0.102 [Approved]
openjdk13 package files upgrade completed. Performing other installation steps.
ERROR: This package does not support 64 bit architecture.
The upgrade of openjdk13 was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\openjdk13\tools\chocolateyinstall.ps1'.
 See log for details.
```
```
2020/09/15 00:39:01 [burp.free.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install burp.free.fireeye
```
```
WSL 2 installation is incomplete.
```
```
amass.docker.fireeye v3.0.25.14
amass.docker.fireeye package files upgrade completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: time="2020-09-15T00:52:57+01:00" level=error msg="Can't add file \\\\?\\C:\\Users\\Commando\\AppData\\Local\\ChocoCache\\docker-build-git021012507\\.git\\hooks\\pre-receive.sample to tar: io: read/write on closed pipe"
The upgrade of amass.docker.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\amass.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - amass.docker.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\amass.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.
Chocolatey reported an unsuccessful exit code of -1. See C:\Users\Commando\AppData\Local\Boxstarter\boxstarter.log for details.
+ Boxstarter finished Calling Chocolatey to install amass.docker.fireeye. This may take several minutes to complete... 00:00:09.6627723
+ Boxstarter finished Calling Chocolatey to install amass.docker.fireeye. This may take several minutes to complete... 00:00:09.6627723


ExitCode: -1
ExitCode: -1
2020/09/15 00:53:00 [amass.docker.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install amass.docker.fireeye
2020/09/15 00:53:00 [amass.docker.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install amass.docker.fireeye
2020/09/15 00:53:00 [amass.docker.fireeye] chocolateyinstall.ps1 [+] INFO : Attempting install of ASREPRoast.fireeye
2020/09/15 00:53:00 [amass.docker.fireeye] chocolateyinstall.ps1 [+] INFO : Attempting install of ASREPRoast.fireeye
WARNING: [!] Installing using host choco.exe! Errors are ignored. Please check to confirm ASREPRoast.fireeye is installed properly
```
```
spiderfoot.docker.fireeye v0.0.0.8
spiderfoot.docker.fireeye package files upgrade completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: time="2020-09-15T01:08:33+01:00" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: open //./pipe/docker_engine: The system cannot find the file specified."
The upgrade of spiderfoot.docker.fireeye was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\spiderfoot.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - spiderfoot.docker.fireeye (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\spiderfoot.docker.fireeye\tools\chocolateyinstall.ps1'.
 See log for details.
Chocolatey reported an unsuccessful exit code of -1. See C:\Users\Commando\AppData\Local\Boxstarter\boxstarter.log for details.
+ Boxstarter finished Calling Chocolatey to install spiderfoot.docker.fireeye. This may take several minutes to complete... 00:00:12.7220965
+ Boxstarter finished Calling Chocolatey to install spiderfoot.docker.fireeye. This may take several minutes to complete... 00:00:12.7220965


ExitCode: -1
ExitCode: -1
2020/09/15 01:08:35 [spiderfoot.docker.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install spiderfoot.docker.fireeye
2020/09/15 01:08:35 [spiderfoot.docker.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install spiderfoot.docker.fireeye
2020/09/15 01:08:35 [spiderfoot.docker.fireeye] chocolateyinstall.ps1 [+] INFO : Attempting install of SpoolerScanner.fireeye
2020/09/15 01:08:35 [spiderfoot.docker.fireeye] chocolateyinstall.ps1 [+] INFO : Attempting install of SpoolerScanner.fireeye
```
```
openvpn.fireeye
By upgrading you accept licenses for the packages.
openvpn.fireeye is not installed. Installing...
[NuGet] Attempting to resolve dependency 'common.fireeye '.
Progress: Downloading openvpn.fireeye 2.4.6.20190317... 96%                                                                                                                                                                 openvpn.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
openvpn.fireeye package files upgrade completed. Performing other installation steps.
The upgrade of openvpn.fireeye was NOT successful.
openvpn.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - openvpn.fireeye (exited 1) - openvpn.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
Chocolatey reported an unsuccessful exit code of 1. See C:\Users\Commando\AppData\Local\Boxstarter\boxstarter.log for details.
+ Boxstarter finished Calling Chocolatey to install openvpn.fireeye. This may take several minutes to complete... 00:02:51.6056389
+ Boxstarter finished Calling Chocolatey to install openvpn.fireeye. This may take several minutes to complete... 00:02:51.6056389


ExitCode: 1
ExitCode: 1
2020/09/15 01:15:58 [SharpSploit.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install openvpn.fireeye
2020/09/15 01:15:58 [SharpSploit.fireeye] chocolateyinstall.ps1 [+] ERROR : Failed to install openvpn.fireeye
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Citrix Binary\Citrix Binary.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Citrix Binary cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Citrix Installed Components\Citrix Installed Components.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Citrix Installed Components cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\EnvVariables\EnvVariables.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\EnvVariables cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Event Logs\Application Event Log\Application Event Log.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Event Logs\Application Event Log cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Event Logs\System Event Log\System Event Log.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Event Logs\System Event Log cannot be removed because it is not empty.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Event Logs cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Microsoft HotFix\Microsoft HotFix.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Microsoft HotFix cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\BNIStack Parameters\BNIStack Parameters.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\BNIStack Parameters cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\BNNS Parameters\BNNS Parameters.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\BNNS Parameters cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Citrix\Citrix.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Citrix cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\CtxHttp\CtxHttp.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\CtxHttp cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\EdgeSight\EdgeSight.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\EdgeSight cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Explorer\Explorer.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Explorer cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\FileSystem\FileSystem.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\FileSystem cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Flash\Flash.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Flash cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-Microsoft-Cryptography\HKLM-Software-Microsoft-Cryptography.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-Microsoft-Cryptography cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-PoliciesExplorer\HKLM-Software-PoliciesExplorer.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-PoliciesExplorer cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-Winlogon\HKLM-Software-Winlogon.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-Winlogon cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-WinNTWindows\HKLM-Software-WinNTWindows.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-Software-WinNTWindows cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-Citrix\HKLM-System-Citrix.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-Citrix cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-CurrentControlSet-Control-TS\HKLM-System-CurrentControlSet-Control-TS.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-CurrentControlSet-Control-TS cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-PrintEnvironment\HKLM-System-PrintEnvironment.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-PrintEnvironment cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-SessionManager\HKLM-System-SessionManager.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\HKLM-System-SessionManager cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Internet Explorer\Internet Explorer.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Internet Explorer cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\LanmanServer Parameters\LanmanServer Parameters.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\LanmanServer Parameters cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\LanmanWorkstation Parameters\LanmanWorkstation Parameters.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\LanmanWorkstation Parameters cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\mrxsmb Parameters\mrxsmb Parameters.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\mrxsmb Parameters cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Print Providers\Print Providers.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Print Providers cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Session Manager Memory Management\Session Manager Memory Management.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Session Manager Memory Management cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Windows\Windows.csv: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry\Windows cannot be removed because it is not empty.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\Registry cannot be removed because it is not empty.
Cannot remove item C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo\EnvInfo.log: You do not have sufficient access rights to perform this operation.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132\EnvInfo cannot be removed because it is not empty.
Directory C:\Users\Commando\AppData\Local\ChocoCache\CTXReceiverInstallLogs-20200915-001132 cannot be removed because it is not empty.
Directory C:\Users\Commando\AppData\Local\ChocoCache cannot be removed because it is not empty.
```
```
Upgrading the following packages:
commandovm.win10.config.fireeye
By upgrading you accept licenses for the packages.
commandovm.win10.config.fireeye is not installed. Installing...
[NuGet] Attempting to resolve dependency 'common.fireeye '.
Progress: Downloading commandovm.win10.config.fireeye 2.0.0.7... 99%                                                                                                                                                        commandovm.win10.config.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
commandovm.win10.config.fireeye package files upgrade completed. Performing other installation steps.
The upgrade of commandovm.win10.config.fireeye was NOT successful.
commandovm.win10.config.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - commandovm.win10.config.fireeye (exited 1) - commandovm.win10.config.fireeye not installed. An error occurred during installation:
 Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
Chocolatey reported an unsuccessful exit code of 1. See C:\Users\Commando\AppData\Local\Boxstarter\boxstarter.log for details.
+ Boxstarter finished Calling Chocolatey to install commandovm.win10.config.fireeye. This may take several minutes to complete... 00:10:44.5115539
+ Boxstarter finished Calling Chocolatey to install commandovm.win10.config.fireeye. This may take several minutes to complete... 00:10:44.5115539
0
Only an exit code of non-zero will fail the package by default. Set
 `--failonstderr` if you want error messages to also fail a script. See
 `choco -h` for details.
 The install of myinstaller was successful.
  Software installed to 'C:\Program Files\NTCore\Explorer Suite\'

Chocolatey installed 1/1 packages. 0 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
Chocolatey : Chocolatey reported an unsuccessful exit code of 1. See C:\Users\Commando\AppData\Local\Boxstarter\boxstarter.log for details.
At C:\ProgramData\boxstarter\BoxStarter.Chocolatey\Invoke-ChocolateyBoxstarter.ps1:199 char:5
+     Chocolatey install $bootstrapPackage -source $source -force:$forc ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,chocolatey
```

**Comment by day1player** [comment]
@DeeBee20 is Docker up and running?

**Comment by DeeBee20** [comment]
@day1player nope had the issues which i listed above while carrying out the installation, i am going to try a different profile install now tonight, which profile has all the tools but no docker invovled.

**Comment by day1player** [comment]
@DeeBee20 what is your host machine? Mac, Windows, Linux?

**Comment by DeeBee20** [comment]
@day1player MacBook Pro 2019 running VMware Fusion

**Comment by DeeBee20** [comment]
@day1player this should not make an issue though right, as i have the VM installing isolated from the base OS

**Comment by day1player** [comment]
> MacBook Pro 2019 running VMware Fusion

Good, you should have no issue 😄

**Comment by DeeBee20** [comment]
@day1player so which profile has all the tools but does not have docker with in the installation process.

**Comment by day1player** [comment]
@DeeBee20 what you could do is use the full profile, but edit the `full.json` file (`./Profiles/full.json`) to remove all of the docker packages (lines 109, 122, and 202), save the file and then install with the same command `.\install.ps1 -profile_file .\Profiles\full.json`

What I would recommend doing (if you dont want Docker) is just to do the default install. Install time will be decreased, and it is less prone to errors. If there are more tools you would like you can install them with choco or by using git. 

Unfortunately based on the error output above for spiderfoot and amass I cant see where the script errored out. Unfortunately it looks like it just didnt print the error output.

If you do want to install Docker (I highly recommend especially if you plan on using Covenant) Make sure in your VMware settings for the VM you enable hypervisor applications in the VM under the Processors and Memory settings:
<img width="627" alt="image" src="https://user-images.githubusercontent.com/37482972/93265150-066c4700-f75d-11ea-8081-3cfb30df0aee.png">

And then I would recommend just using the nestedv profile: `\install.ps1 -profile_file .\Profiles\full.json`

As for the commando config package, just keep an eye out to make sure the background changes. If the background doesnt change after a "successful" install, you will have to reinstall with `cinst -y commandovm.win10.config.fireeye -f` (#150 )

**Comment by DeeBee20** [comment]
@day1player cool i will give the nestedv a try now

**Comment by DeeBee20** [comment]
@day1player i keep getting this error in the installation as well

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/61301688/93313593-97482e80-f800-11ea-976a-30786b306614.png">

**Comment by day1player** [comment]
@DeeBee20 that is normal, I am still working out a way to get that going manually.. Please check out #152

**Comment by day1player** [comment]
If you do not wish to use WSL 2 you can also just go into the Docker settings and disable WSL2. It will work perfectly normal with WSL1

**Comment by DeeBee20** [comment]
@day1player hahahaha, i have had the VM at this state 3-4 times now and because i have been getting that error, i restarted the process thinking there was faults lol

**Comment by day1player** [comment]
Oh no lol.. No biggie, you live and you learn!

**Comment by DeeBee20** [comment]
@day1player think i've finally cracked it ....

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/61301688/93403607-e71a0a80-f87f-11ea-85fc-358248ef4b18.png">

<img width="1135" alt="image" src="https://user-images.githubusercontent.com/61301688/93403620-f305cc80-f87f-11ea-9b84-ca60854b41e5.png">

**Comment by day1player** [comment]
@DeeBee20 excellent. Looks pretty good to me! Any other issues?

**Comment by DeeBee20** [comment]
@day1player Final piece of the puzzle implemented. 

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/61301688/93516269-b5598000-f921-11ea-9f7f-ed852f3f6db8.png">

Thank you for all your help.

**Comment by day1player** [comment]
@DeeBee20 thats great, if you ever need help with Covenant there is also a channel in the Slack.. if you dont want to post your email here you can always email me at commandovm@fireeye.com and I can request you an invite.

**Comment by DeeBee20** [comment]
@day1player daniel.brooks34@icloud.com can you send me an invite now please so i can delete this message off

**Comment by day1player** [comment]
invite requested, feel free to delete
